### PR TITLE
Allowing fields comparison in grid filters

### PR DESCRIPTION
**Pull request by Roshyo**
| Q               | A
| --------------- | -----
| Branch?         | 1.0
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Adding a support to fields comparison. So we can filter for example an entity related to an other one.
There is still the PHPCRODM to verify (marked as TODO: ) since I don't know if that works

@pamil @CoderMaggie  : I believe it's the right branch this time (master for features :) )

### CODE REVIEW

File: src/Sylius/Component/Grid/Data/ExpressionBuilderInterface.php [hunk 1]
```diff
@@ -54,6 +54,22 @@ public function equals(string $field, $value);
      */
     public function notEquals(string $field, $value);
 
+    /**
+     * @param string $field
+     * @param string $secondField
+     *
+     * @return mixed
+     */
+    public function fieldsEquals(string $field, string $secondField);
```
Comment (pamil) [inline src/Sylius/Component/Grid/Data/ExpressionBuilderInterface.php]: Adding a method to an interface is a BC break unfortunately :/

Comment (Roshyo) [reply src/Sylius/Component/Grid/Data/ExpressionBuilderInterface.php]: Oh, indeed... What would be the best to do it then ?

**Comment by Roshyo** [comment]
Do not stale ?

How to proceed ? do we have to wait until 2.0 for this feature ?

**Comment by vvasiloi** [comment]
Isn't it possible to implement this use-case with a custom filter and `Sylius\Bundle\GridBundle\Doctrine\ORM\ExpressionBuilder::comparison`?

**Comment by Zales0123** [comment]
Hello Stephane, it's been a while. In the meantime, **SyliusGridBundle** was extracted to the [separate repository](https://github.com/Sylius/SyliusGridBundle/), so I suppose it should be implemented there. Thank you! 🖖

**Pull request closed without merging**